inject/src/main/java/io/micronaut/inject/writer/BeanDefinitionWriter.java

```diff
@@ -424,6 +513,13 @@
     private boolean constructorRequiresReflection;
     private boolean disabled = false;
 
+    private final boolean keepConfPropInjectPoints;
+
+    {
+        String property = System.getProperty("micronaut.keep-conf-prop-inject-points");
+        keepConfPropInjectPoints = property != null && property.toLowerCase(Locale.ROOT).equals("true");
```

Comment:
I don't think we should be using system properties here, if anything they should be annotation processor inputs.

However in this case I think we can do something simpler and just have `@ConfigurationProperties(metadata=false)` to disable build time metadata or true to include it since this applies only to configuration propeties. This will probably need to be a meta annotation with `@AliasFor` pointing to `@ConfigurationReader` and also be added to `@EachProperty` so it is consistent.

We can then perhaps debate whether this should default to disable metadata or not. Personally I don't think anyone is using this metadata in real world apps, but it is perhaps a breaking change.

Reply:
I think it would be better to have a custom annotation because it’s more of internal implementation.

Reply:
Perhaps, but if it is a custom annotation then we would have to make it work for other cases and not just configuration properties, like for example if I don't need the injection points for regular `@Inject` fields / methods